### PR TITLE
Code Quality: Implement Splash Screen

### PR DESCRIPTION
**Pull request by 0x5bfa**
## Code Quality: Refactor App.App class

Before start to work on splash screen, I obliged to make those changes because I unconsiously will do in the splash screen introduction PR.

### Motivation & Context

- Improved readability

> **Warning**<br/>Please do not review code lines that I did not changed. I cannot make more diff. If there're more codes that are preferred to change, I will do that and mark that PR as the final one for that stage.

### PR Checklist

- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #4180
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [ ] Are there any other steps that were used to validate these changes?

### Screenshots

_None_

### CODE REVIEW

File: src/Files.App/Program.cs [hunk 1]
```diff
@@ -13,20 +12,31 @@
...
 	internal class Program
 	{
-		// Note:
-		//  We can't declare Main to be async because in a WinUI app
-		//  This prevents Narrator from reading XAML elements
-		//  https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/Samples/AppLifecycle/Instancing/cs-winui-packaged/CsWinUiDesktopInstancing/CsWinUiDesktopInstancing/Program.cs
-		//  STAThread has no effect if main is async, needed for Clipboard
+		private const uint CWMO_DEFAULT = 0;
+
+		private const uint INFINITE = 0xFFFFFFFF;
```
Comment (yaira2) [inline src/Files.App/Program.cs:22]: Where are these coming from?

Comment (0x5bfa) [reply src/Files.App/Program.cs:22]: Coming from 153-155. Just moved to the top according to the guideline.

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by yaira2** [comment]
@0x5bfa the app seems to be crashing, can you investigate?

File: src/Files.App/MainWindow.xaml.cs [hunk 2]
```diff
@@ -1,77 +1,77 @@
...
-
-			// Set logo
-			AppWindow.SetIcon(Path.Combine(Package.Current.InstalledLocation.Path, App.LogoPath));
+			PersistenceId = "FilesMainWindow";
+			MinHeight = 328;
+			MinWidth = 516;
 
-			// Extend title bar
+			// Initialize AppWindow
+			AppWindow.Title = "Files";
```
Comment (QuaintMako) [inline src/Files.App/MainWindow.xaml.cs:35]: No localization needed here?

File: src/Files.App/Views/SplashScreen.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
```
Comment (QuaintMako) [inline src/Files.App/Views/SplashScreen.xaml.cs:14]: Are all of those really needed?

File: src/Files.App/Views/SplashScreen.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+// To learn more about WinUI, the WinUI project structure,
+// and more about our project templates, see: http://aka.ms/winui-project-info.
```
Comment (QuaintMako) [inline src/Files.App/Views/SplashScreen.xaml.cs:17]: Can be removed

File: src/Files.App/Views/SplashScreen.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace Files.App.Views
+{
+	/// <summary>
+	/// An empty page that can be used on its own or navigated to within a Frame.
+	/// </summary>
```
Comment (QuaintMako) [inline src/Files.App/Views/SplashScreen.xaml.cs:23]: Can be removed

**Review by QuaintMako — COMMENTED** [review]
Very small points, but LGTM otherwise. Cannot test it.

**Comment by mdtauk** [comment]
I have updated the Figma assets for the SplashScreen images, @yaira2 will need to decide if he wants to re-export them at the new Win32 size guidance, to put into the repo.

**Comment by mdtauk** [comment]
image

Red - 96x96 icon
Blue - 310x310 Tile Image

Titlebar area at the top 32px height

**Review by ferrariofilippo — COMMENTED** [review]
I can't test this PR, whenever I try to run it I get this exception while the app is still loading
(Everything works fine on `main`)

```
{--------- UNHANDLED EXCEPTION --------->>>> 
HRESULT: -2146233088--- 
MESSAGE ---A Task's exception(s) were not observed either by Waiting on the Task or accessing its Exception property. 
As a result, the unobserved exception was rethrown by the finalizer thread. 
(Unable to cast object of type 'Microsoft.UI.Xaml.Controls.Grid' to type 'Microsoft.UI.Xaml.Controls.Frame'.)
--- INNER ---
System.InvalidCastException: Unable to cast object of type 'Microsoft.UI.Xaml.Controls.Grid' to type 'Microsoft.UI.Xaml.Controls.Frame'.   at Files.App.MainWindow.InitializeApplication(Object activatedEventArgs)
---------------------------------------}
```

**Pull request closed without merging**